### ISSUE TITLE
AutoNetClassification throwing out of memory exception.

### ISSUE THREAD

**Issue by aimanakheel** [automl/Auto-PyTorch#255]
Hi Folks, 
Hope this finds you well.

I am trying to run the following setup. 

```
X_train, X_test, y_train, y_test = \
            sklearn.model_selection.train_test_split(X,
                                                     y,
                                                     random_state=random_state,
                                                     test_size=0.30,
                                                     stratify=y)

autoPyTorch = AutoNetClassification("tiny_cs",  # config preset
                                    log_level='info',
                                    budget_type='epochs', 
                                    min_budget=1, 
                                    max_budget=9
                                    result_logger_dir="logs/",
                                    categorical_features = cat_col_indx,  #[True, True, False, False]
                                    random_seed = random_state
                                   )

result = autoPyTorch.fit(X_train, y_train, validation_split=0.3)

y_pred = autoPyTorch.predict(X_test)

print("Accuracy score", sklearn.metrics.accuracy_score(y_test, y_pred))
```
However, i am seeing the following exception in the output. This doesn't stop the execution and it continues to the next task. 
```
17:06:57 [AutoNet] Start bohb
17:06:57 DISPATCHER: started the 'discover_worker' thread
17:06:57 WORKER: start listening for jobs
17:06:57 DISPATCHER: started the 'job_runner' thread
17:06:57 DISPATCHER: Pyro daemon running on 172.16.188.199:35719
17:06:57 DISPATCHER: discovered new worker, hpbandster.run_0.worker.ip-172-16-188-199.6045.-1140100637476672
17:06:57 HBMASTER: adjusted queue size to (0, 1)
17:06:57 DISPATCHER: A new worker triggered discover_worker
17:06:57 HBMASTER: starting run at 1623776817.9087155
17:06:57 WORKER: start processing job (0, 0, 0)
17:06:57 Fit optimization pipeline
17:06:58 [AutoNet] CV split 0 of 1
17:08:50 Unable to allocate 69.4 GiB for an array with shape (1376606, 6767) and data type float64
17:08:50 WORKER: registered result for job (0, 0, 0) with dispatcher
17:08:50 WORKER: start processing job (0, 0, 1)
17:08:50 Fit optimization pipeline
17:08:50 [AutoNet] CV split 0 of 1
17:10:43 Unable to allocate 69.4 GiB for an array with shape (1376606, 6767) and data type float64
17:10:43 WORKER: registered result for job (0, 0, 1) with dispatcher
17:10:43 WORKER: start processing job (0, 0, 2)
17:10:43 Fit optimization pipeline
17:10:44 [AutoNet] CV split 0 of 1
17:12:33 Unable to allocate 69.4 GiB for an array with shape (1376606, 6767) and data type float64
17:12:34 WORKER: registered result for job (0, 0, 2) with dispatcher
17:12:34 WORKER: start processing job (0, 0, 3)
17:12:34 Fit optimization pipeline
17:12:34 [AutoNet] CV split 0 of 1
```
Currently my ram capacity is 144 GiB. Is there a workaround for this issue.

**Comment by ravinkohli** [comment]
Hey, thanks for raising this issue. Could you tell me if you have modified the value of n_workers? Ideally, given the data requirement of 69 GiB only two such arrays can be stored in your ram. So I suggest you to set the n_workers to 1 so we have enough memory to also store the model weights essentially making this a sequential optimization. 

Given that your dataset is tabular which is supported in the upcoming release currently in the `development` branch, I encourage you to give that a try. You can find the documentation [here](https://automl.github.io/Auto-PyTorch/development/). It's a completely redefined API based on the sklearn API and improves the robustness of the underlying code with proper exception handling. With this version, you can set the n_jobs to 1 directly when initializing the [`Task`](https://automl.github.io/Auto-PyTorch/development/api.html)

**Comment by aimanakheel** [comment]
After installing via the `development` branch im getting the following exception during import. 
```
from autoPyTorch.api.tabular_classification import TabularClassificationTask

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-470db6f31052> in <module>
----> 1 from autoPyTorch.api.tabular_classification import TabularClassificationTask
      2 
      3 from utilities import pickle_it, get_pickle
      4 import pandas as pd
      5 from sklearn.preprocessing import LabelEncoder

~/anaconda3/envs/mxnet_latest_p37/lib/python3.7/site-packages/autoPyTorch-0.1.0-py3.7.egg/autoPyTorch/api/tabular_classification.py in <module>
      7 import pandas as pd
      8 
----> 9 from autoPyTorch.api.base_task import BaseTask
     10 from autoPyTorch.automl_common.common.utils.backend import Backend
     11 from autoPyTorch.constants import (

~/anaconda3/envs/mxnet_latest_p37/lib/python3.7/site-packages/autoPyTorch-0.1.0-py3.7.egg/autoPyTorch/api/base_task.py in <module>
     29 from smac.tae import StatusType
     30 
---> 31 from autoPyTorch.automl_common.common.utils.backend import Backend, create
     32 from autoPyTorch.constants import (
     33     REGRESSION_TASKS,

ModuleNotFoundError: No module named 'autoPyTorch.automl_common'

```

**Comment by ravinkohli** [comment]
Oh, this points to a discrepancy in the documentation. I'm sorry for that. Could you please re install using the following steps:

```
# Following commands assume the user is in a cloned directory of Auto-Pytorch

# We also need to initialize the automl_common repository as follows
# You can find more information about this here:
# https://github.com/automl/automl_common/
git submodule update --init --recursive

# Create the environment
conda create -n autopytorch python=3.8
conda activate autopytorch
conda install swig
cat requirements.txt | xargs -n 1 -L 1 pip install
python setup.py install
```

**Comment by ravinkohli** [comment]
I think I gave the wrong solution to your problem however, we have fixed the issue you are facing with PR #324.
